File: 0500-keyboard-row/0500-keyboard-row.java
class Solution {
     Map<Integer, Set<Character>> m = new HashMap<>();

    void put(int k, String s) {
        Set<Character> charSet = new HashSet<>();
        for (char ch : s.toCharArray()) {
            charSet.add(ch);
        }
        m.put(k, charSet);
    }

    public String[] findWords(String[] w) {
        String s1 = "qwertyuiop", s2 = "asdfghjkl", s3 = "zxcvbnm";
        put(1, s1);
        put(2, s2);
        put(3, s3);

        List<String> ans = new ArrayList<>();
        for (String x : w) {
            boolean found = false;
            for (int i = 1; i <= 3; i++) {
                int fl = 1;
                for (char ch : x.toCharArray()) {
                    ch = Character.toLowerCase(ch); // Convert char to lowercase
                    if (!m.get(i).contains(ch)) {
                        fl = 0;
                        break;
                    }
                }
                if (fl == 1) {
                    ans.add(x);
                    found = true;
                    break;
                }
            }
        }
        return ans.toArray(new String[0]);
    }
}

File: 0500-keyboard-row/0500-keyboard-row.cpp
class Solution {
public:
    unordered_map< int, unordered_set<char> > m;
    void put( int k,string s){
        for( char ch :s){
            m[k].insert(ch);
        }
    }
    vector<string> findWords(vector<string>& w) {
        
        
        
        string s1="qwertyuiop" , s2="asdfghjkl" , s3="zxcvbnm";
        
        put(1,s1);
        put(2,s2);
        put(3,s3);
        
        vector< string > ans;
        for( string x:w){
            
            for( int i=1;i<=3;i++){
                int fl=1;
                for( char ch :x){
                    ch=ch | 32;
                    if( m[i].find(ch)==m[i].end() ){
                        fl=0;
                        break;
                    }
                }
                if(fl){
                    ans.push_back(x);
                    break;
                }
            }
        }
        return ans;
    }
};